### PR TITLE
Added metadata preset for Sign4

### PR DESCRIPTION
**Pull request by dcylabs**

### CODE REVIEW

**Comment by mdwhatcott** [comment]
@dcylabs - Can you tell us more about what you're trying to accomplish?

**Comment by dcylabs** [comment]
@mdwhatcott - I try using Sign4 behind CloudFlare, where the guess logic of service and region can't work. With this PR you can hard-pass a preset of metadata that could not be guessed

File: awsauth.go [hunk 1]
```diff
@@ -172,15 +175,52 @@ func (this *Credentials) expired() bool {
...
+type Metadata struct {
+	Algorithm       string
+	CredentialScope string
+	SignedHeaders   string
+	Date            string
+	Region          string
+	Service         string
+}
+
+func (m *Metadata) SetAlgorithm(value string){
```
Comment (mdwhatcott) [inline awsauth.go]: I see that these setter functions prevent any original value from being overwritten--what's the need for that constraint?

Comment (mdwhatcott) [inline awsauth.go]: I'm wondering why we need setter functions at...

**Comment by mdwhatcott** [comment]
Due to #49 I am closing all pull requests.

**Pull request closed without merging**